applications/targets/tests/test_services.py

```diff
@@ -0,0 +1,79 @@
+from django.test import TestCase
+import mock
+
+from . import factories
+from applications.targets.services import TargetMatchingService
+from applications.users.tests.factories import UserFactory
+
+
+class TargetMatchingServiceTest(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        self.topic = factories.TopicFactory()
+        self.target = factories.default_target(self.user, self.topic)
+
+    def call_matching_service(self):
+        matching = TargetMatchingService()
+        matching.process_target(self.target)
+
+    @mock.patch(
+        'applications.notifications.services.NotificationCreator.create',
+        return_value=None
+    )
+    def test_targets_do_not_create_notificaitons(self, notificator):
+        self.call_matching_service()
+
+        notificator.assert_not_called()
+
+    @mock.patch(
+        'applications.notifications.services.NotificationCreator.create',
+        return_value=None
+    )
+    def test_compatible_targets_create_notificaitons(self, notificator):
+        matching_target = factories.compatible_in_radius_target(self.topic)
+        matching_target2 = factories.compatible_in_radius_target2(self.topic)
+
+        self.call_matching_service()
+
+        self.assertEqual(notificator.call_count, 4)
+
+        calls = [
+            mock.call(
+                matching_target.owner,
+                {'target_id': self.target.id},
+                'You have a new match',
+                f'A {self.target.topic.name} target is near you: {self.target.title}',
+            ),
```

Comment:
Maybe you could use a constant for the messages and format them when necessary. That way In case you need to update the message text you don't have to do it in multiple places. 

The "You have a new match." is constant and for the second part you could define `NOTIFICATION_BODY = A {} target is near you: {}` and then you use `NOTIFICATION_BODY.format(self.target.topic.name, self.target.title`)